lektor_s3.py

```diff
@@ -142,14 +155,53 @@ def compute_diff(self, local, remote):
                 diff['delete'].append(k)
         return diff
 
+    def create_headers(self, filename):
+        """ Set file headers from mimetype and config. """
+        mime, _ = mimetypes.guess_type(filename)
+        if mime is None:
+            mime = "text/plain"
+        headers = {'ContentType': mime}
+
+        # Set defaults first
+        for upload_arg in TransferManager.ALLOWED_UPLOAD_ARGS:
+            upload_arg_value = self.config.defaults().get(upload_arg.lower())
```

Comment:
I would suggest that you use the [inifile](https://github.com/mitsuhiko/python-inifile) library rather than ConfigParser. Although its undocumented, I believe that @mitsuhiko wrote it because he was unhappy with the idiosyncrasies of the ConfigParser module. [Lektor already depends on this library](https://github.com/lektor/lektor/blob/master/setup.py#L23), and uses it to parse all the ini files that are in a Lektor project.